Makefile

```diff
@@ -0,0 +1,121 @@
+# The `go env GOPATH` will work for all cases for Go 1.8+.
+GOPATH            ?= $(shell go env GOPATH)
+
+TMP_GOPATH        ?= /tmp/thanos-go
+GOBIN             ?= $(firstword $(subst :, ,${GOPATH}))/bin
+GOPROXY           ?= https://proxy.golang.org
+GIT               ?= $(shell which git)
+
+# Version from 12.12.2019
+GOJSONTOYAML_VERSION    ?= bf2969bbd742d117a9524b859fb417fefb67565d
+GOJSONTOYAML            ?= $(GOBIN)/gojsontoyaml-$(GOJSONTOYAML_VERSION)
+
+# Version from 9.02.2020. We use https://github.com/google/go-jsonnet for ease of building.
+GO_JSONNET_VERSION         ?= 31b9ace0f65a5a5717ed9521b57d8d9e18f24070
+GO_JSONNET                 ?= $(GOBIN)/jsonnet-$(GO_JSONNET_VERSION)
+# v0.2.0
+JSONNET_BUNDLER_VERSION ?= 184841238bb5df1b886ccaecd3e179bc55c17687
+JSONNET_BUNDLER         ?= $(GOBIN)/jb-$(JSONNET_BUNDLER_VERSION)
+
+# fetch_go_bin_version downloads (go gets) the binary from specific version and installs it in $(GOBIN)/<bin>-<version>
+# arguments:
+# $(1): Install path. (e.g github.com/campoy/embedmd)
+# $(2): Tag or revision for checkout.
+# TODO(bwplotka): Move to just using modules, however make sure to not use or edit Thanos go.mod file!
+define fetch_go_bin_version
+	@mkdir -p $(GOBIN)
+	@mkdir -p $(TMP_GOPATH)
+
+	@echo ">> fetching $(1)@$(2) revision/version"
+	@if [ ! -d '$(TMP_GOPATH)/src/$(1)' ]; then \
+    GOPATH='$(TMP_GOPATH)' GO111MODULE='off' go get -d -u '$(1)/...'; \
+  else \
+    CDPATH='' cd -- '$(TMP_GOPATH)/src/$(1)' && git fetch; \
+  fi
+	@CDPATH='' cd -- '$(TMP_GOPATH)/src/$(1)' && git checkout -f -q '$(2)'
+	@echo ">> installing $(1)@$(2)"
+	@GOBIN='$(TMP_GOPATH)/bin' GOPATH='$(TMP_GOPATH)' GO111MODULE='off' go install '$(1)'
+	@mv -- '$(TMP_GOPATH)/bin/$(shell basename $(1))' '$(GOBIN)/$(shell basename $(1))-$(2)'
+	@echo ">> produced $(GOBIN)/$(shell basename $(1))-$(2)"
+
+endef
+
+help: ## Displays this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n\nTargets:\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-10s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+# deps ensures jsonnet deps are installed.
+.PHONY: deps
+deps: ## Install libsonnet dependencies using jsonnet-bundler.
+deps: $(JSONNET_BUNDLER) jsonnetfile.json jsonnetfile.lock.json
+	@echo ">> install jsonnet deps via jb"
+	$(JSONNET_BUNDLER) install
+
+JSONNET_FMT := jsonnetfmt -n 2 --max-blank-lines 2 --string-style s --comment-style s -i
+
+.PHONY: format
+format: ## Auto-formats jsonnet files.
+	@which jsonnetfmt 2>/dev/null || ( \
+		echo "Cannot find jsonnetfmt command, please install from https://github.com/google/jsonnet/releases. If your C++ does not support GLIBCXX_3.4.20, please use xxx-in-container target like format-in-container." && exit 1 \
+	)
+
+	@echo ">> formatting jsonnet"
+	@find . -type f -not -path './vendor/*' \( -name '*.libsonnet' -o -name '*.jsonnet' \) | xargs -n 1 -- $(JSONNET_FMT)
+
+.PHONY: gen
+gen: ## Generates manifests for all environments from jsonnet.
```

Comment:
Just for sake of consistency between repos, should we rename this as `generate`?

Reply:
:+1: